add appveyor

enricosada opened the pull request:
ref https://github.com/dotnet/cli/issues/2113

Download and unzip .NET Core SDK Binaries in a subdirectory `.netcoresdk` and add that to PATH

Easy to understand what do, easy to reproduce locally.

I used powershell 3, with 5 was a bit more readable as oneliner but not every image support it (not this image from appveyor anyway)

/cc @blackdwarf

enricosada committed:
add appveyor

enricosada committed:
dotnet core sdk preview1 is already installed in appveyor

enricosada commented:
updated, appveyor now has '.net core sdk preview1' already installed in the build worker image.
i left commented the code for install a specific version, if it's needed (useful to try development versions). but i can remove it, np

richlander commented:
Thanks for this.

Help me a bit to understand. I'm starting with zero knowledge on this, so be easy with me ;)

My questions:

- This is just the build script. How will people know how to use this? I'm thinking that an associated help doc explaining this would be very useful.
- Does this file need to be in the root of the repo? If we had a samples directory, could it be in there?
- Will you update this when it needs updating?

enricosada commented:
> This is just the build script. How will people know how to use this? I'm thinking that an associated help doc explaining this would be very useful.

People need that as example how to install .net core sdk in appveyor, not how to do build.
Everyone has different build script (fake, rake, cake, msbuild, bat, powershell), but all they need is the .net core sdk installed and in PATH, actual command for build (restore, pack, etc) are up to developer. 
But if they already use appveyor, that's standard.

> Does this file need to be in the root of the repo? If we had a samples directory, could it be in there?

Yes by design of appveyor that's the default where it's expected (and where developer search it). 
Also, if you enable appveyor for this repo, we can test this script every commit for free, and also (if you want), can build all the examples in this repo.

> Will you update this when it needs updating?

sure, i use .net core sdk for open source project, that's really easy to update it

enricosada commented:
The only question open i think is if you want to use msi installer or zip binaries.
I think installer has pro because it's short as install step, but zip are nice because i can test it locally without fear of changes in my machine (i really love that about .net core sdk)

richlander commented:
Makes sense.

I think all we need is a doc about using .NET Core in CI systems. Is that the right frame?

Yes, we are definitely believer in the zip option, for the reason you give. You have high confidence on the safety of your machine that way.

enricosada commented:
there is a page in docs about ci (with appveyor/travis TODO), [Using .NET Core SDK and tools in Continuous Integration (CI)
](https://github.com/dotnet/core-docs/blob/master/docs/core-concepts/core-sdk/cli/using-ci-with-cli.md), i am pretty sure i seen it published time ago in the dotnet.github.com website too

blackdwarf commented:
@enricosada @richlander yep, the doc exists since day 1 of RC2. Just the way we roll. Enrico has graciously offered to write up Travis and AppVeyor samples which he did. 

However, these build scripts will not build this repo, they will be in the docs, so their contents should be included in the relevant doc.

richlander commented:
Sounds good. As long as we have some content that explain how and why to use these files, that's great.

blackdwarf commented:
@richlander nope, we've decided to actually put them just like this into the root of the repo and point people to them with a small note that says "explanation left for lunch. Will be back presently." Or something like that. 

:)))

richlander commented:
@blackdwarf Oh perfect. Exactly what I was hoping for. I know our readers enjoy being left in suspense.

blackdwarf commented:
@richlander of course, every literary masterpiece needs a twist, right? :)

blackdwarf commented:
@enricosada can you include these scripts as well in the relevant doc (http://dotnet.github.io/docs/core-concepts/core-sdk/cli/using-ci-with-cli.html)?

richlander commented:
@enricosada Can you add some content?

enricosada commented:
yes, i'll fix doc sry for delay

enricosada committed:
update docs

enricosada commented:
ok, rebased, added some content to docs + link to example.

i added an example of build matrix because helps a lot to test multiple version of .NET Core SDK  (or just simplify configuration). Like that is easy to test prerelease asp.net core or preview2

blackdwarf commented:
LGTM. @mairaw are we ok with merging this? This is a pretty strong addition to the tools docs. 

@enricosada as always, thank you kind sir for your stellar work and contributions! 😄 🎉

mairaw commented on docs/core-concepts/core-sdk/cli/using-ci-with-cli.md:
````diff
@@ -53,6 +53,33 @@ The below sections show examples of configurations using the mentioned CI SaaS o
+os: Visual Studio 2015
+```
+
+It's possibile to install a specific version of .NET Core SDK, see [example appveyor.yml](https://github.com/dotnet/core-docs/blob/master/appveyor.yml) 
````
possibile -> possible

mairaw commented on docs/core-concepts/core-sdk/cli/using-ci-with-cli.md:
```diff
@@ -53,6 +53,33 @@ The below sections show examples of configurations using the mentioned CI SaaS o
 **TODO**
```
line 18: if its not -> if it's not

mairaw commented on docs/core-concepts/core-sdk/cli/using-ci-with-cli.md:
```diff
@@ -53,6 +53,33 @@ The below sections show examples of configurations using the mentioned CI SaaS o
+It's possibile to install a specific version of .NET Core SDK, see [example appveyor.yml](https://github.com/dotnet/core-docs/blob/master/appveyor.yml) 
+for more info. 
+
+In the example the .NET Core SDK binaries are downloaded, unzipped in a subdirectory and added to `PATH` env var
```
nit: add comma after in the example and a period in the end of the sentence

mairaw commented on docs/core-concepts/core-sdk/cli/using-ci-with-cli.md:
```diff
@@ -53,6 +53,33 @@ The below sections show examples of configurations using the mentioned CI SaaS o
+In the example the .NET Core SDK binaries are downloaded, unzipped in a subdirectory and added to `PATH` env var
+
+A build matrix can be added to run integration tests with multiple version of 
+the .NET Core SDK
```
nit: missing punctuation

mairaw commented:
Cool. Left a couple of comments @blackdwarf, @enricosada

cartermp commented:
This LGTM as well, modulo @mairaw's comments.  Thanks @enricosada!

enricosada committed:
improvements from feedback

enricosada commented:
done, thx @mairaw for review.
